app/uk/gov/hmrc/minorentityidentificationfrontend/models/TrustDetails.scala

```diff
@@ -71,4 +72,35 @@ object TrustDetails {
     registrationBlock ++ utrBlock ++ saPostcodeBlock ++ chrnBlock ++ businessVerificationBlock ++ identifiersMatchBlock
   }
 
+  def writesForAudit(optTrustDetails: Option[TrustDetails], businessVerificationCheck: Boolean): JsObject = {
+    optTrustDetails match {
+      case Some(trustDetails) =>
+        val optSaUtrBlock = trustDetails.optUtr match {
+          case Some(saUtr) => Json.obj("SAUTR" -> saUtr.value)
+          case None => Json.obj()
+        }
+
+        val optSaPostCodeBlock = trustDetails.optSaPostcode match {
+          case Some(saPostCode) => Json.obj("SApostcode" -> saPostCode)
+          case None => Json.obj()
+        }
+
+        val optCHRNBlock = trustDetails.optChrn match {
+          case Some(chrn) => Json.obj("CHRN" -> chrn)
+          case None => Json.obj()
+        }
+
+        Json.obj(
+          "isMatch" -> defineAuditIdentifiersMatch(trustDetails.optIdentifiersMatch),
+          "VerificationStatus" -> defineAuditBusinessVerificationStatus(trustDetails.optBusinessVerificationStatus, businessVerificationCheck),
+          "RegisterApiStatus" -> defineAuditRegistrationStatus(trustDetails.optRegistrationStatus)
+        ) ++ optSaUtrBlock ++ optSaPostCodeBlock ++ optCHRNBlock
+      case None =>
+        Json.obj(
```

Comment:
I'm not certain this code would get executed, but if it did passing None to defineAuditBusinessVerificationStatus would cause problems.